soh/soh/z_message_OTR.cpp

```diff
@@ -15,7 +15,40 @@ extern "C" MessageTableEntry* sFraMessageEntryTablePtr;
 extern "C" MessageTableEntry* sStaffMessageEntryTablePtr;
 //extern "C" MessageTableEntry* _message_0xFFFC_nes;	
 
-MessageTableEntry* OTRMessage_LoadTable(const char* filePath, bool isNES) {
+static void SetMessageEntry(MessageTableEntry& entry, const LUS::MessageEntry& msgEntry) {
+    entry.textId = msgEntry.id;
+    entry.typePos = (msgEntry.textboxType << 4) | msgEntry.textboxYPos;
+    entry.segment = msgEntry.msg.c_str();
+    entry.msgSize = msgEntry.msg.size();
+}
+
+static void OTRMessage_LoadCustom(const std::string& folderPath, MessageTableEntry*& table, size_t tableSize) {
+    auto lst = *LUS::Context::GetInstance()->GetResourceManager()->GetArchive()->ListFiles(folderPath).get();
+
+    for (auto& tPath : lst) {
+        auto file = std::static_pointer_cast<LUS::Text>(LUS::Context::GetInstance()->GetResourceManager()->LoadResource(tPath));
+
+        for (size_t j = 0; j < file->messages.size(); ++j) {
+            // Check if same text ID exists already
+            size_t existingIndex = 0;
+            while (existingIndex < tableSize && table[existingIndex].textId != file->messages[j].id) {
+                ++existingIndex;
+            }
+
+            if (existingIndex < tableSize) {
+                // Replace existing message
+                SetMessageEntry(table[existingIndex], file->messages[j]);
+            } else {
+                // Add new message
+                table = (MessageTableEntry*)realloc(table, sizeof(MessageTableEntry) * (tableSize + 1));
+                SetMessageEntry(table[tableSize], file->messages[j]);
+                ++tableSize;
+            }
```

Comment:
When we get to the point having of brand new messages that aren't overriding a vanilla text ID packed into OTRs, we'll want to use our `CustomMessageTable`s rather than extending the vanilla tables. So for now just get rid of this part entirely and stick to having this just override the vanilla tables.